### PR TITLE
Add osx to travis

### PR DESCRIPTION
**Pull request by hayd**
Unfortunately this times-out on osx...

I am not sure what can be done to speed up the build script.
I tried with -j8 and without -j. They didn't seem to make much difference.

https://travis-ci.org/hayd/deno/jobs/419484578
It got _seemingly_ close to:
```
[907/1054] CXX obj/third_party/v8/v8_base/liftoff-assembler.o
```
Note: Machine seems to be the [same size as linux](https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system).

#563

### CODE REVIEW

**Comment by ry** [comment]
Super hacky - but this is what we've been resorting to:

 You need to do a partial build. Change travis to only build the target v8_shell, and then exit successfully without running the tests.  This will populate the cache. The undo the changes and run again.

File: .travis.yml [hunk 1]
```diff
@@ -23,6 +26,9 @@ install: |
   if [ $? != 0 ]; then
     curl -sSf https://sh.rustup.rs | sh -s -- -y
   fi
+  if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then
+    brew install ccache && export PATH="/usr/local/opt/ccache/libexec:$PATH";
```
Comment (ry) [inline .travis.yml]: It doesn't have it already?

Comment (hayd) [reply .travis.yml]: apparently not: https://docs.travis-ci.com/user/caching/#ccache-on-os-x

**Comment by ry** [comment]
@hayd I see you're poking at this - I really appreciate it - it's tedious work. Maybe try `./tools/build.py  third_party/v8:v8_base` instead of v8_shell - that's a smaller target to built. If all else fails I can upgrade travis to pro - which gives 50m longer timeout...

**Comment by hayd** [comment]
@ry looks like that did the trick! 😎

**Comment by ry** [comment]
@hayd Rad! Nice work!

**Comment by ry** [comment]
@hayd I can already tell it's going to timeout because the ccache is empty https://travis-ci.com/denoland/deno/jobs/141767781
```
0.03s$ ccache -s
cache directory                     /Users/travis/.ccache
primary config                      /Users/travis/.ccache/ccache.conf
secondary config      (readonly)    /usr/local/Cellar/ccache/3.4.2/etc/ccache.conf
cache hit (direct)                     0
cache hit (preprocessed)               0
cache miss                             0
cache hit rate                      0.00 %
cleanups performed                     0
files in cache                         0
cache size                           0.0 kB
max cache size                       5.0 GB
```
I think the solution - it pains me to say - is to land a commit in master which builds v8_base only on Mac.... Then land another where we remove that conditional... 
Maybe a slightly better solution is to build v8_base and skip tests if ccache is empty?

**Comment by hayd** [comment]
Yep, sorry I should've left the two commits to do in sequence. I wonder if it would've used this cache doing a "merge pull request". 

I'll look into adding a conditional test, it should be okay (test the size of the .ccache directory).

**Comment by ry** [comment]
Yeah, merge might work.

**Comment by hayd** [comment]
I wonder if this can be fixed in Travis (this time).

1. Set the default branch to this one (pr/570) if possible, or push a commit with v8base to another branch and set to that.
2. retry the build.
3. turn the default back to master.

It looks like the cache is populated from a list of possible places, and default branch is the last one.

**Pull request merged**